source/app/libpd.c

```diff
@@ -171,14 +179,41 @@ static void parodus_receive()
                         wrp_free_struct (res_wrp_msg);
                     }
             }
+
+            //handle cloud-status retrieve response received from parodus
+            if (wrp_msg->msg_type == WRP_MSG_TYPE__RETREIVE)
+            {
+				sourceService = wrp_get_msg_element(WRP_ID_ELEMENT__SERVICE, wrp_msg, SOURCE);
+				sourceApplication = wrp_get_msg_element(WRP_ID_ELEMENT__APPLICATION, wrp_msg, SOURCE);
+
+				if(sourceService != NULL && sourceApplication != NULL && strcmp(sourceService,"parodus")== 0 && strcmp(sourceApplication,"cloud-status")== 0)
+				{
+					WalInfo("cloud-status Retrieve response received from parodus : %s len %lu\n",(char *)wrp_msg->u.crud.payload, strlen(wrp_msg->u.crud.payload) );
+
+					parsePayloadForStatus(wrp_msg->u.crud.payload, &status);
+					if(status !=NULL)
+					{
+						//set this as global conn status. adding lock before update it.
+						pthread_mutex_lock (&cloud_mut);
```

Comment:
Add lock inside the setter and getter functions for global cloud_status

Reply:
Incorporated.